pytransform3d/trajectories.py

```diff
@@ -436,6 +515,322 @@ def pqs_from_dual_quaternions(dqs):
+    outer_if_inner_else_mask = np.logical_and(outer_if_mask,
+                                              np.logical_not(inner_if_mask))
+
+    # Initialize the outputs
+    qs = np.zeros((dqs.shape[0], 3))
+    thetas[outer_if_mask] = ds[outer_if_mask]
+    hs = np.full(dqs.shape[0], np.inf)
+
+    if np.any(outer_if_inner_if_mask):
+        s_axis[outer_if_inner_if_mask] = np.array([1, 0, 0])
+
+    if np.any(outer_if_inner_else_mask):
+        s_axis[outer_if_inner_else_mask] = (
+            translation[outer_if_inner_else_mask]
+            / ds[outer_if_inner_else_mask][..., np.newaxis]
+        )
+
+    qs = np.zeros((dqs.shape[0], 3))
+    thetas[outer_if_mask] = ds[outer_if_mask]
+    hs = np.full(dqs.shape[0], np.inf)
+
+    if np.any(outer_else_mask):
+        distance = np.einsum('ij,ij->i',
+                             translation[outer_else_mask],
+                             s_axis[outer_else_mask])
+
+        moment = 0.5 * (
+            np.cross(translation[outer_else_mask], s_axis[outer_else_mask]) +
+            (translation[outer_else_mask] - distance[..., np.newaxis] *
+             s_axis[outer_else_mask])
+            / np.tan(0.5 * thetas[outer_else_mask])[..., np.newaxis]
+        )
+
+        qs[outer_else_mask] = np.cross(s_axis[outer_else_mask], moment)
+        hs[outer_else_mask] = distance / thetas[outer_else_mask]
+
+    return qs, s_axis, hs, thetas
+
+
+def dual_quaternions_from_screw_parameters(qs, s_axis, hs, thetas):
+    """Compute dual quaternions from arrays of screw parameters.
+
+    Parameters
+    ----------
+    qs : array-like, shape (...,3)
+        Vector to a point on the screw axis
+
+    s_axis : array-like, shape (...,3)
+        Direction vector of the screw axis
+
+    hs : array-like, shape (...,)
+        Pitch of the screw. The pitch is the ratio of translation and rotation
+        of the screw axis. Infinite pitch indicates pure translation.
+
+    thetas : array-like, shape (...,)
+        Parameter of the transformation: theta is the angle of rotation
+        and h * theta the translation.
+
+    Returns
+    -------
+    dqs : array, shape (...,8)
+        Unit dual quaternion to represent transform:
+        (pw, px, py, pz, qw, qx, qy, qz)
+
+    See Also
+    --------
+    pytransform3d.transformations.dual_quaternion_from_screw_parameters :
+        Compute dual quaternion from screw parameters.
+    """
+    ds = np.zeros_like(hs)
+
+    h_is_inf_mask = np.isinf(hs)
+    h_is_not_inf_mask = np.logical_not(h_is_inf_mask)
+
+    mod_thetas = thetas.copy()
+    if np.any(h_is_inf_mask):
+        ds[h_is_inf_mask] = thetas[h_is_inf_mask]
+        mod_thetas[h_is_inf_mask] = 0
+
+    if np.any(h_is_not_inf_mask):
+        ds[h_is_not_inf_mask] = hs[h_is_not_inf_mask] * \
+            thetas[h_is_not_inf_mask]
+
+    moments = np.cross(qs, s_axis)
+    half_distances = 0.5 * ds
+    half_thetas = 0.5 * mod_thetas
+    sin_half_angles = np.sin(0.5 * half_thetas)
```

Comment:
This should now be an error. We compute `sin(0.5 * 0.5 * mod_thetas)`, it was `sin(0.5 * mod_thetas)` before

Reply:
So it seems like there is also no unit test that checks this function properly.